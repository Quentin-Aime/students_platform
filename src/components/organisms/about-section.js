import React from "react";
import styled from "styled-components";
import { Header } from "../molecules/header";
import { SubTitle } from "../atoms/subtitle";
import { Title } from "../atoms/title";
import { Picture } from "../atoms/picture";
import { Text } from "../atoms/text";
import group from "../../assets/images/group.jpg";
import { Layout } from "../atoms/layout";
import { Grid } from "../atoms/grid";
import { TitleContainer } from "../molecules/title-container";

export const AboutSection = ({ member, setMember }) => {
  return (
    <Layout id="about">
      <AboutContainer>
        <HeaderContainer>
          <Header setMember={setMember} />
        </HeaderContainer>
        <TitleContainer>
          <div>
            <StyledSubtitle
              content={member === 1 ? "OUR STORY, OUR GROUP." : "MY STORY"}
              align="left"
            />
          </div>
          <div>
            <Title
              content={member === 1 ? "about us" : "about me"}
              align="left"
            />
          </div>
        </TitleContainer>
        <Grid>
          <div style={{ width: "100%" }}>
            <Picture imgUrl={group} />
          </div>
          <div style={{ width: "100%" }}>
            <Text style={{ width: "70%" }}>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut at
              egestas risus. Nam dignissim tellus et quam viverra bibendum.
              Praesent vel mi vitae sem efficitur porta vel sit amet orci. Proin
              quam purus, lobortis quis porttitor in, consequat maximus elit.
              Quisque interdum nunc quis suscipit tempus. Fusce tempus ex in
              enim sagittis hendrerit. Nunc odio elit, volutpat sed nunc sed,
              lobortis rhoncus nulla. Phasellus vitae metus sed ligula ornare
              vehicula. Sed finibus mollis commodo.
            </Text>
          </div>
        </Grid>
      </AboutContainer>
    </Layout>
  );
};

const AboutContainer = styled.section`
  width: 100%;
`;

const HeaderContainer = styled.div`
  margin: 0 auto;
  width: 80%;
  display: flex;
  justify-content: center;
  margin-bottom: 5rem;
`;

const StyledSubtitle = styled(SubTitle)`
  margin: 2rem 0;
`;
