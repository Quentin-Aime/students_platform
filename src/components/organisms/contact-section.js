import React from "react";
import styled from "styled-components";
import { SubTitle } from "../atoms/subtitle";
import { Title } from "../atoms/title";
import ig from "../../assets/images/instagram.svg";
import twitter from "../../assets/images/twitter.svg";
import github from "../../assets/images/github.svg";
import gmail from "../../assets/images/gmail.svg";
import { Social } from "../molecules/social";

export const ContactSection = () => {
  const socials = [
    {
      label: "instagram",
      logo: ig,
      link: "",
    },
    {
      label: "twitter",
      logo: twitter,
      link: "",
    },
    {
      label: "gmail",
      logo: gmail,
      link: "",
    },
    {
      label: "github",
      logo: github,
      link: "",
    },
  ];
  return (
    <ContactContainer>
      <SubTitle content="DO YOU WANT TO CONTACT US?" />
      <Title content="contact us" />
      <LinksContainer>
        {socials.map((social) => (
          <CustomSocial
            label={social.label}
            logo={social.logo}
            link={social.link}
          />
        ))}
      </LinksContainer>
    </ContactContainer>
  );
};

const ContactContainer = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

const LinksContainer = styled.div`
  width: 100%;
  justify-content: center;
  display: flex;
  flex-wrap: wrap;
`;

const CustomSocial = styled(Social)`
  margin-right: 1rem;
`;
