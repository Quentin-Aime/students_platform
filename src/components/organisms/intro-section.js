import React from "react";
import styled from "styled-components";
import { PageTitle } from "../atoms/title";
import { SubTitle } from "../atoms/subtitle";
import backgroundPicture from "../../assets/images/noodle_bg.png";
import { Layout } from "../atoms/layout";

export const IntroSection = () => {
  return (
    <section id="home">
      <Layout>
        <IntroContainer>
          <div className="text">
            <SubTitle content="GROUP 5 - SCHOOL PROJECT" />
            <PageTitle content="creating a new project" alignment="left" />
          </div>
          <img src={backgroundPicture} alt="ramen drawing" />
        </IntroContainer>
      </Layout>
    </section>
  );
};

const IntroContainer = styled.div`
  position: relative;
  width: auto;
  height: calc(100vh - 110px);
  display: flex;
  align-items: center;
  justify-content: center;
  flex-wrap: no-wrap;
  .text {
    position: relative;
    z-index: 1;
  }
  img {
    position: relative;
    margin-left: -300px;
  }
`;
