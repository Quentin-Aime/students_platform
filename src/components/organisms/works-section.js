import React from "react";
import styled from "styled-components";
import work1 from "../../assets/images/work-1.jpg";
import work2 from "../../assets/images/work-2.jpg";
import { WorkCard } from "../molecules/work-card";
import { color } from "../../assets/const";
import { Layout } from "../atoms/layout";
import { Grid } from "../atoms/grid";

export const WorksSections = ({ member, setMember }) => {
  return (
    <WorkContainer id="works">
      <Layout>
        <Grid>
          <WorkCard imgUrl={work1} title="01 : Lorem Ipsum" />
          <WorkCard imgUrl={work2} title="02 : Lorem Ipsum" />
        </Grid>
      </Layout>
    </WorkContainer>
  );
};

const WorkContainer = styled.section`
  width: 100%;
  height: 90vh;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background: ${color.secondary};
`;
