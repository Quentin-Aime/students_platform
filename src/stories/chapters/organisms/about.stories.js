import React from "react";
import { storiesOf } from "@storybook/react";
import { AboutSection } from "../../../components/organisms/about-section";

export const doc = `
\`\`\`
import { AboutSection } from "../../../components/organisms/about-section";
\`\`\`
`;

storiesOf("Organisms|AboutSection", module)
  .addParameters({
    readme: {
      content: doc,
    },
  })
  .add("default", () => {
    return <AboutSection />;
  });
