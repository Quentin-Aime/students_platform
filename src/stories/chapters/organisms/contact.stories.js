import React from "react";
import { storiesOf } from "@storybook/react";
import { ContactSection } from "../../../components/organisms/contact-section";

export const doc = `
\`\`\`
import { ContactSection } from "../../../components/organisms/contact-section";
\`\`\`
`;

storiesOf("Organisms|ContactSection", module)
  .addParameters({
    readme: {
      content: doc,
    },
  })
  .add("default", () => {
    return <ContactSection />;
  });
