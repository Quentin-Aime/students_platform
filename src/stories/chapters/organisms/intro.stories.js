import React from "react";
import { storiesOf } from "@storybook/react";
import { IntroSection } from "../../../components/organisms/intro-section.js";

export const doc = `
\`\`\`
import { IntroSection } from "../../../components/organisms/intro-section.js";
\`\`\`
`;

storiesOf("Organisms|IntroSection", module)
  .addParameters({
    readme: {
      content: doc,
    },
  })
  .add("default", () => {
    return <IntroSection />;
  });
