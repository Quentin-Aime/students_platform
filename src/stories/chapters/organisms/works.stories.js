import React from "react";
import { storiesOf } from "@storybook/react";
import { WorksSections } from "../../../components/organisms/works-section";

export const doc = `
\`\`\`
import { WorksSections } from "../../../components/organisms/works-section";
\`\`\`
`;

storiesOf("Organisms|WorksSections", module)
  .addParameters({
    readme: {
      content: doc,
    },
  })
  .add("default", () => {
    return <WorksSections />;
  });
