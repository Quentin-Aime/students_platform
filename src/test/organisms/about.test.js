import React from "react";
import { create } from "react-test-renderer";
import { AboutSection } from "../../components/organisms/about-section";

describe("About section component", () => {
  test("Matches the snapshot", () => {
    const about = create(<AboutSection />);
    expect(about.toJSON()).toMatchSnapshot();
  });
});
