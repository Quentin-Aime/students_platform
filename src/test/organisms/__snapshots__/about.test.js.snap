// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About section component Matches the snapshot 1`] = `
<section
  className="sc-Axmtr iyERoH"
>
  <div
    className="sc-AxirZ jYjzEF"
  >
    <div
      className="sc-AxjAm kufRBV"
    >
      <img
        alt="thumbnail"
        src="ramen.svg"
      />
    </div>
    <div
      className="sc-AxjAm kufRBV"
    >
      <img
        alt="thumbnail"
        src="melanie.svg"
      />
    </div>
    <div
      className="sc-AxjAm kufRBV"
    >
      <img
        alt="thumbnail"
        src="marie.svg"
      />
    </div>
    <div
      className="sc-AxjAm kufRBV"
    >
      <img
        alt="thumbnail"
        src="aymeline.svg"
      />
    </div>
    <div
      className="sc-AxjAm kufRBV"
    >
      <img
        alt="thumbnail"
        src="quentin.svg"
      />
    </div>
    <div
      className="sc-AxjAm kufRBV"
    >
      <img
        alt="thumbnail"
        src="rainald.svg"
      />
    </div>
    <div
      className="sc-AxjAm kufRBV"
    >
      <img
        alt="thumbnail"
        src="federica.svg"
      />
    </div>
    <div
      className="sc-AxjAm kufRBV"
    >
      <img
        alt="thumbnail"
        src="mathieu.svg"
      />
    </div>
    <div
      className="sc-AxjAm kufRBV"
    >
      <img
        alt="thumbnail"
        src="ketsia.svg"
      />
    </div>
  </div>
  <h3
    className="sc-AxiKw dXvIgp"
  >
    MY STORY
  </h3>
  <h2
    className="sc-AxhUy bJGCUG"
  >
    about me
  </h2>
  <div
    className="sc-AxmLO eTgygq"
  >
    <div
      className="sc-AxgMl clkmuO"
    >
      <img
        alt="background"
        src="group.jpg"
      />
    </div>
    <div
      className="sc-AxheI ldjAXd"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut at egestas risus. Nam dignissim tellus et quam viverra bibendum. Praesent vel mi vitae sem efficitur porta vel sit amet orci. Proin quam purus, lobortis quis porttitor in, consequat maximus elit. Quisque interdum nunc quis suscipit tempus. Fusce tempus ex in enim sagittis hendrerit. Nunc odio elit, volutpat sed nunc sed, lobortis rhoncus nulla. Phasellus vitae metus sed ligula ornare vehicula. Sed finibus mollis commodo.
    </div>
  </div>
</section>
`;
